main.go

```diff
@@ -92,16 +101,26 @@ func main() {
 			fmt.Printf("\n\n")
 			continue
 		}
-
+		ctxLogger, _ := getContextLogger()
+		requestID := uid.NewV4().String()
+		ctxLogger = ctxLogger.With(zap.String("RequestID", requestID))
+		sess, _, err := provider_util.OpenProviderSession(conf, providerRegistry, providerName, ctxLogger)
+		if err != nil {
+			ctxLogger.Error("Failed to get session", zap.Reflect("Error", err))
+			continue
+		}
+		defer sess.Close()
+		defer ctxLogger.Sync()
 		if choiceN == 1 {
 			fmt.Println("You selected choice to get volume details")
 			fmt.Printf("Please enter volume ID: ")
 			_, er11 = fmt.Scanf("%s", &volumeID)
 			volume, errr := sess.GetVolume(volumeID)
 			if errr == nil {
-				logger.Info("SUCCESSFULLY get volume details ================>", zap.Reflect("VolumeDetails", volume))
+				ctxLogger.Info("SUCCESSFULLY get volume details ================>", zap.Reflect("VolumeDetails", volume))
 			} else {
-				logger.Info("FAILED to get volume details ================>", zap.Reflect("VolumeID", volumeID), zap.Reflect("Error", errr))
+				errr = updateRequestID(errr, requestID)
```

Comment:
`errr = updateRequestID(errr, requestID)`  - why we are updating requestId explicitly? (edited) 
I guess , `volume, errr := sess.GetVolume(volumeID)` should return error with `request id` 
All the library API should return error with resuestID. So driver code  do not have to worry about that . The driver will just pass in requestID to API  through sesssion
Also , the error object belongs to provider package(volume-provider/vpc) , So let all the properties of error be set from provider only .